server/pruning.go

```diff
@@ -36,3 +43,74 @@ func GetPruningOptionsFromFlags(appOpts types.AppOptions) (pruningtypes.PruningO
 		return pruningtypes.PruningOptions{}, fmt.Errorf("unknown pruning strategy %s", strategy)
 	}
 }
+
+// PruningCmd prunes the sdk root multi store history versions based on the pruning options
+// specified by command flags.
+func PruningCmd(providerCreator types.StoreProviderCreator) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "prune",
+		Short: "prune history stetes based on the pruning options specified by flags",
+		Long: `Pruning options can be provided via the '--pruning' flag or alternatively with '--pruning-keep-recent', and
+		'pruning-interval' together.
+		
+		For '--pruning' the options are as follows:
+		
+		default: the last 362880 states are kept, pruning at 10 block intervals
+		nothing: all historic states will be saved, nothing will be deleted (i.e. archiving node)
+		everything: 2 latest states will be kept; pruning at 10 block intervals.
+		custom: allow pruning options to be manually specified through 'pruning-keep-recent', and 'pruning-interval'
+		`,
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			serverCtx := GetServerContextFromCmd(cmd)
```

Comment:
Why use `serverCtx` at all? Should be no need. Just get everything you need directly from the flags.

Reply:
we need to bind flags value to `serverCtx.Viper` by calling:
```go
if err := serverCtx.Viper.BindPFlags(cmd.Flags()); err != nil {
				return err
			}
```
and then we can easily get pruning options from `serverCtx.Viper` :
```go
pruningOptions, err := GetPruningOptionsFromFlags(serverCtx.Viper)
```

Reply:
This seems so silly and unnecessary to me. Just create a new local viper instance and bind that. No need for `serverCtx`.

Reply:
now use a local `viper` to bind flag values instead.